File: IDATT2106 Systemutvikling 2 med smidig prosjekt/backend/src/main/java/com/ntnu/gidd/model/PasswordResetToken.java
package com.ntnu.gidd.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.time.ZonedDateTime;

@Getter
@Setter
@Entity
public class PasswordResetToken extends UUIDModel {
	
	//60 minutes
	private static final int EXPIRATION = 60;
	
	@OneToOne(targetEntity = User.class, fetch = FetchType.EAGER)
	@JoinColumn(nullable = false, referencedColumnName = "id")
	private User user;
	
	private ZonedDateTime expirationDate;
	
	public PasswordResetToken() {
		expirationDate = ZonedDateTime.now().plusMinutes(EXPIRATION);
	}
}

File: IDATT2105 Full-stack applikasjonsutvikling/Rombestilling/backend/src/main/kotlin/ntnu/idatt2105/security/token/PasswordResetToken.kt
package ntnu.idatt2105.security.token

import ntnu.idatt2105.user.model.User
import java.time.ZonedDateTime
import java.util.*
import javax.persistence.*

@Entity
data class PasswordResetToken(
    @Id @Column(columnDefinition = "CHAR(32)")
    var id: UUID = UUID.randomUUID(),
    @OneToOne
    @JoinColumn(nullable = true, referencedColumnName = "id")
    val user: User,
    val expirationDate: ZonedDateTime = ZonedDateTime.now().plusMinutes(EXPIRATION.toLong())
) {

    companion object {
        // 60 minutes
        private const val EXPIRATION = 60
    }
}
fun PasswordResetToken.isAfter() = this.expirationDate.isAfter(ZonedDateTime.now())
